lib/overcommit/hook/pre_push/protected_branches.rb

```diff
@@ -0,0 +1,27 @@
+module Overcommit::Hook::PrePush
+  # Prevents destructive updates to specified branches.
+  class ProtectedBranches < Base
+    def run
+      return :pass unless illegal_pushes.any?
+
+      messages = illegal_pushes.map do |pushed_ref|
+        "Deleting or force-pushing to #{pushed_ref.remote_ref} is not allowed."
+      end
+
+      [:fail, messages.join("\n")]
+    end
+
+    private
+
+    def branches
+      @branches ||= config['branches']
+    end
+
+    def illegal_pushes
+      @illegal_pushes ||= pushed_refs.select do |pushed_ref|
+        (pushed_ref.deleted? || pushed_ref.forced?) &&
+          branches.any? { |branch| pushed_ref.remote_ref =~ /#{branch}/ }
```

Comment:
Are you sure you want to do a regex match here? Suppose I have a branch called `master-and-commander`?